aten/src/ATen/core/Dict_inl.h

```diff
@@ -120,9 +120,9 @@ template<class Key_, class Value_>
 std::pair<typename Dict<Key, Value>::iterator, bool> Dict<Key, Value>::insert(Key_&& key, Value_&& value) const {
   static_assert(std::is_constructible<Key, Key_>::value, "Wrong type for the key argument of Dict::insert");
   static_assert(std::is_constructible<Value, Value_>::value, "Wrong type for the value argument of Dict::insert");
-  auto inserted = impl_->dict.insert(std::pair<IValue, IValue>{
-    Key(std::forward<Key_>(key)),
-    Value(std::forward<Value_>(value))});
+  auto inserted = impl_->dict.emplace(
```

Comment:
```suggestion
  auto inserted = impl_->dict.try_emplace(
```
Pretty sure this is what you intended. This will avoid creating the the std::pair entirely if it's already in the dict.

Reply:
nope, that's most definitely not what I intended ;) There are at least 2 reasons for this:
1) `try_emplace` exists only in broken implementations of dict like the ones that are in std. Such function makes no sense in non-broken implementations that aten is using https://github.com/pytorch/pytorch/blob/main/c10/util/order_preserving_flat_hash_map.h#L537-L553
2) the implementation from aten already skips pair creation if key is present